system/modules/core/library/Contao/Files/Php.php

```diff
@@ -57,6 +57,9 @@ public function rmdir($strDirectory)
 	 */
 	public function fopen($strFile, $strMode)
 	{
+		if (is_link($strFile)) {
+			$strFile = readlink($strFile);
```

Comment:
@JanMalte: in your test case you only use absolute paths, if I'm correct. For a relative symlink (ln -s ../b/file file)  ``readlink`` also  returns only the relative value (says doc, verified on Ubuntu/PHP-5.5 and cygwin/PHP-5.6, not yet tested on Windows with native symlinks - Win7ff). Thus you must evaluate it to an absolute path in the context of the link.

Generally, I feel a bit uncomfortable with manually resolving a symlink and would rather leave the job to the OS when possible. With ``fopen/fwrite`` this is not a problem at all, both ``fopen(...'w')`` and ``fopen(...'w+')`` write to the linked file and leave the link in proper state (verified as above). With some paranoia, one could use ``fopen(...'w+')`` for an existing symlink to make sure the old file is used, but is is not necessary.

So the minimal invasive operation would be, in my opinion, to change only the rename function to evaluate symlinks - or (if possible, e.g. for config) to use ``fopen/fwrite`` instead of ``mv`` in case of an existing symlink (I know that this is no longer atomic but I wonder if that will be an issue in normal operation).

My minimal test code:
```
<?php
// run test in separate subdir /var/www/test (e.g.) to avoid simplicity effects
// (cd /var/www;echo "hi there">bin/x.txt;ln -sf bin/x.txt x;cd test;php test.php)
$link_rel = '../x';
$link_abs = '/var/www/x';
$file_rel = readlink($link_rel);
$file_abs = readlink($link_abs);
printf("%s=>%s - %s=>%s\n", $link_rel, $file_rel, $link_abs, $file_abs);
// get absolute path
if ($file_abs[0] != '/') {
    // may not work with native Windows paths...
    $file_abs = dirname($link_abs) . '/' . $file_abs;
    printf("abs result: %s\n", $file_abs);
}
// verify linked file is written, not the link
echo file_get_contents($file_abs);
$fp=fopen($link_rel, 'w');
fwrite($fp, "changed there\n");
fclose($fp);
if (is_link($link_rel)) {
    echo "still a link\n";
}
echo file_get_contents($file_abs);
```
The output:
```
www-data@heimdall:~/test$ php x2.php
../x=>bin/x.txt - /var/www/x=>bin/x.txt
abs result: /var/www/bin/x.txt
hi there
still a link
changed there
```